rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderTest.java

```diff
@@ -2149,6 +2150,435 @@ public void saveCoinBaseInformation_after_RSKIP143() throws IOException {
+    }
+
+    @Test
+    public void getActiveFederationCreationBlockHeight_after_fork() {
+        Repository repository = mock(Repository.class);
+        // If by chance the repository is called I want to force the tests to fail
+        when(repository.getStorageBytes(PrecompiledContracts.BRIDGE_ADDR, DataWord.fromString("activeFedCreationBlockHeight"))).thenReturn(new byte[] { 1 });
+
+        BridgeStorageProvider provider0 = new BridgeStorageProvider(
+                repository, PrecompiledContracts.BRIDGE_ADDR,
+                config.getNetworkConstants().getBridgeConstants(), activationsAllForks
+        );
+
+        assertEquals(Long.valueOf(1L), provider0.getActiveFederationCreationBlockHeight());
+
+        // If the network upgrade is not enabled we shouldn't be reading the repository
+        verify(repository, atLeastOnce()).getStorageBytes(PrecompiledContracts.BRIDGE_ADDR, DataWord.fromString("activeFedCreationBlockHeight"));
+    }
+
+    @Test
+    public void setActiveFederationCreationBlockHeightAndGetActiveFederationCreationBlockHeight() {
+        Repository repository = createRepository();
+        Repository track = repository.startTracking();
+
+        BridgeStorageProvider provider0 = new BridgeStorageProvider(
+                track, PrecompiledContracts.BRIDGE_ADDR,
+                config.getNetworkConstants().getBridgeConstants(), activationsAllForks
+        );
+
+        // We store the value
+        provider0.setActiveFederationCreationBlockHeight(1L);
+        provider0.saveActiveFederationCreationBlockHeight();
+        track.commit();
+
+        track = repository.startTracking();
+
+        BridgeStorageProvider provider = new BridgeStorageProvider(
+                track, PrecompiledContracts.BRIDGE_ADDR,
+                config.getNetworkConstants().getBridgeConstants(), activationsAllForks
+        );
+
+        // And then we get it back
+        assertThat(provider.getActiveFederationCreationBlockHeight(), is(1L));
+    }
+
+    @Test
+    public void saveActiveFederationCreationBlockHeight_after_RSKIP186() {
+        Repository repository = mock(Repository.class);
+
+        BridgeStorageProvider provider0 = new BridgeStorageProvider(
+                repository, PrecompiledContracts.BRIDGE_ADDR,
+                config.getNetworkConstants().getBridgeConstants(), activationsAllForks
+        );
+
+        provider0.setActiveFederationCreationBlockHeight(10L);
+        provider0.saveActiveFederationCreationBlockHeight();
+
+        // Once the network upgrade is active, we will store it in the repository
+        verify(repository, times(1)).addStorageBytes(
+                PrecompiledContracts.BRIDGE_ADDR,
+                DataWord.fromString("activeFedCreationBlockHeight"),
+                BridgeSerializationUtils.serializeLong(10L)
+        );
+    }
+
+    @Test
+    public void saveActiveFederationCreationBlockHeight_before_RSKIP186() {
```

Comment:
```suggestion
    public void getActiveFederationCreationBlockHeight_before_RSKIP186() {
```

Reply:
@Vovchyk in this test you are not saving, just getting. Seems like a duplicate from a previous test or am I wrong?

Reply:
thanks @josedahlquist. Good catch. Actually in this class we have two `getXXX` tests (before n after hf), two `saveXXX` (before n after hf) and one `saveXXXAndGetXXX` (which tests saving to and getting from non-mocked repository). I've fixed `saveXXX_before_RSKIP186` test that you're refering to above in here ce43150